uaclient/contract.py

```diff
@@ -171,6 +173,90 @@ def report_machine_activity(self):
             machine_token["activityInfo"] = response
             self.cfg.write_cache("machine-token", machine_token)
 
+    def get_magic_attach_token_info(self, magic_token: str) -> Dict[str, Any]:
+        """Request magic attach token info.
+
+        When the magic token is registered, it will contain new fields
+        that will allow us to know that the attach process can proceed
+        """
+        headers = self.headers()
+        headers.update({"Authorization": "Bearer {}".format(magic_token)})
+        url = API_V1_MAGIC_ATTACH
+
+        try:
+            response, _ = self.request_url(url, headers=headers)
+        except exceptions.UrlError as e:
+            if isinstance(e, exceptions.ContractAPIError):
+                if hasattr(e, "code"):
+                    if e.code == 401:
```

Comment:
For all endpoints that accept a "Bearer <magic-attach-token>" kind of auth, `401` always means one of: not found, invalid format (e.g., someone said the token is "abc123") or expired already.

The message from ua-contracts will not distinguish between these cases, and it will just say "unauthorized".

I'd suggest: `exceptions.MagicAttachTokenError` with the following message to the user:
> The magic attach token is invalid, has expired or never existed.